monocle/utils.py

```diff
@@ -54,18 +54,87 @@ class Detector(object):
     tests_regexp = ".*[Tt]est.*"
     tests_re = re.compile(tests_regexp)
 
+    issue_tracker_links = {
```

Comment:
do we want to support other bug trackers like Jira or bugzilla? Enterprise GitHub?

Reply:
Yes. We will be able to complete with others issue trackers in SAS mode like jira.altasian. In a second step the idea is to make the regexps configurable when we will have the project feature. The Detector will take a project config in input.  But let keep it simple for a first step.

Reply:
I was thinking something simple like detect url with jira|bug|issue in them.

Reply:
altassian.net would not match in that case.

Reply:
but I could probably add a generic url matcher too.

Reply:
Hum at query level adding to much regexp (so increasing complexity) is forbidden by lucene:
```
"index":"monocle.test.monocle-unittest","caused_by":{"type":"too_complex_to_determinize_exception","reason":"Determinizing .*https?:\\/\\/.*issue.*|https?:\\/\\/.*bug.*|https?:\\/\\/.*jira.*|https?:\\/\\/.*toto.*|https?:\\/\\/.*titi.*.*|.* \\#[0-9]+.*|.*[^/ :]+\\/[^/]+\\#[0-9]+.*|.*GH-[1-9]+.*|.*https?:\\/\\/.+.atlassian.net\\/browse\\/.*.* would result in more than 10000 states.","caused_by":{"type":"too_complex_to_determinize_exception","reason":"Determinizing automaton with 111 states and 415 transitions would result in more than 10000 states."}}}}]},"status":400}
```

We should probably remove the has_issue_tracker_links query parameter. For the moment it does not make sense to support every style during a query. Nobody will have issue tracker on multiple tracker. So we could re-enable it when we will have the project definition support where we will be able to say my project: issue-tracker: github.com or altassian.net or whatever.

So my proposal is to keep that the change is enhanced with the has_issue_tacker_links attribute. But remove the has_issue_tracker_links query parameter.

What do you think ?

Reply:
Or the has_issue_tracker_links query parameter could contains choices: generic, github.com, altassian.net So the number of regexp is limited to a known set.

Reply:
I've implemented my last proposal to prevent having to complex regexp queries.

Reply:
I think it's a good time to start having this in a specific section in the configuration with a regexp per index.